src/react.ts

```diff
@@ -15,7 +15,7 @@ import {
 
 import { getVersion, subscribe, snapshot } from './vanilla'
 import { createMutableSource, useMutableSource } from './useMutableSource'
-import type { NonPromise } from './vanilla'
+import type { DeepResolveType } from './vanilla'
```

Comment:
Actually, it's a great type name, I don't have any problem with it, but could it be shorter?

Reply:
`DeepResolve`? `ResolveType`?
`DeepResolveType` feels better as it looks special.

Reply:
Yea, no problem with that, let's go with it.

Reply:
It's a complex name for a complex utility type 😂

And by complex I mean special like @dai-shi said.